packages/block-library/src/file/index.php

```diff
@@ -19,6 +19,28 @@ function render_block_core_file( $attributes, $content ) {
 		wp_enqueue_script( 'wp-block-file-view' );
 	}
 
+	// Update object's aria-label attribute if present in block HTML.
+
+	// Match an aria-label attribute from an object tag.
+	$pattern = '@<object.+(?<attribute>aria-label="(?<filename>[^"]+)?")@i';
+	$content = preg_replace_callback(
+		$pattern,
+		function ( $matches ) {
+			$filename     = ! empty( $matches['filename'] ) ? $matches['filename'] : '';
+			$has_filename = ! empty( $filename );
+			$label        = $has_filename ?
+				sprintf(
+					/* translators: %s: filename. */
+					__( 'Embed of %s.' ),
+					$filename
+				)
+				: __( 'PDF embed' );
+
+			return str_replace( $matches['attribute'], sprintf( 'aria-label="%s"', $label ), $matches[0] );
+		},
+		$content
+	);
+
```

Comment:
thanks for noticing the work on the `WP_HTML_Walker` - for reference, here is roughly the code that would replace what we have here…

```php
// Update object's aria-label attribute if present in block HTML.

// Match an aria-label attribute from an object tag.
$w = new WP_HTML_Walker( $content );
if ( $w->next_tag( 'object' ) ) {
	$filename = $w->get_attribute( 'aria-label' ) ?: '';
	$label    = empty( $filename )
		sprintf(
			/* translators: %s: filename. */
			__( 'Embed of %s.' ),
			$filename
		)
		: __( 'PDF embed' );

	$w->set_attribute( 'aria-label', $label );
	return $w;
}

return $content;
```

Reply:
@dmsnell Thanks for the code. Do you think we should we create a follow up issue to update the file block when the work on WP_HTML_Walker is merge ?

Reply:
sounds like a great idea!

Reply:
I created the follow-up PR - #60494.